volatility3/framework/plugins/windows/pslist.py

```diff
@@ -187,17 +187,22 @@ def _generator(self):
                 (_, _, offset, _, _) = list(memory.mapping(offset = proc.vol.offset, length = 0))[0]
 
             file_output = "Disabled"
-            if self.config['dump']:
-                file_handle = self.process_dump(self.context, self.config['nt_symbols'], pe_table_name, proc, self.open)
-                file_output = "Error outputting file"
-                if file_handle:
-                    file_handle.close()
-                    file_output = str(file_handle.preferred_filename)
-
-            yield (0, (proc.UniqueProcessId, proc.InheritedFromUniqueProcessId,
-                       proc.ImageFileName.cast("string", max_length = proc.ImageFileName.vol.count, errors = 'replace'),
-                       format_hints.Hex(offset), proc.ActiveThreads, proc.get_handle_count(), proc.get_session_id(),
-                       proc.get_is_wow64(), proc.get_create_time(), proc.get_exit_time(), file_output))
+
+            try:
+                if self.config['dump']:
+                    file_handle = self.process_dump(self.context, self.config['nt_symbols'], pe_table_name, proc, self.open)
+                    file_output = "Error outputting file"
+                    if file_handle:
+                        file_handle.close()
+                        file_output = str(file_handle.preferred_filename)
+
+                yield (0, (proc.UniqueProcessId, proc.InheritedFromUniqueProcessId,
+                   proc.ImageFileName.cast("string", max_length = proc.ImageFileName.vol.count, errors = 'replace'),
+                   format_hints.Hex(offset), proc.ActiveThreads, proc.get_handle_count(), proc.get_session_id(),
+                   proc.get_is_wow64(), proc.get_create_time(), proc.get_exit_time(), file_output))
+
+            except exceptions.InvalidAddressException:
+                vollog.debug("Invalid process found at address: {:x}. Skipping".format(proc.vol.offset))  
```

Comment:
I'm wondering if we should up this from debug to warning?  I don't think most people will turn on the debug flags by default which is the reason for throwing this exception in the first place.

Reply:
warning sounds good.. or even info.. some samples smear early so you only get a few processes listed

Reply:
I chose to try/except in the output yield as its the old issue of list_processes returning invalid process objects. In the past we discussed this making it possible for structures that overlap pages to be partially valid and leaving that as an option to developers, but forcing try/except when accessing structure members as the object may be invalid and referencing invalid members throws the exception. As long as list_processes returns invalid objects then the calling code will need to try/except to avoid backtracing on object access.